### ISSUE TITLE
Links with parenthesis mess up syntax highlighting

### ISSUE THREAD

**Issue by Qix-** [jbt/markdown-editor#35]
```markdown
[some link](https://en.wikipedia.org/wiki/Tree_(data_structure)#Terminologies_used_in_Trees)
```

parses and is displayed correctly in live preview, but the syntax highlighting is incorrect.

<img width="599" alt="screen shot 2015-07-28 at 6 48 35 pm" src="https://cloud.githubusercontent.com/assets/885648/8947529/475a2910-3559-11e5-879d-74510f9120bc.png">

**Comment by jbt** [comment]
This'll be an issue with [CodeMirror's markdown mode](https://github.com/codemirror/CodeMirror/blob/master/mode/markdown/markdown.js). Looks like it's been updated upstream since I last grabbed it so I'll see about updating it and that may sort this.

That said, I really should round off and merge #25 which ditches CodeMirror entirely and would make things nicer to work with

**Comment by jbt** [comment]
Well I bumped to latest and that doesn't seem to have fixed it. I'll have a little dig around and see if it's feasible to try and figure out the necessary changes in codemirror or whether it'd be less effort to finish off that PR

**Comment by Qix-** [comment]
:+1: for prism. Does it properly handle the case I mentioned?

**Comment by jbt** [comment]
Yup, you can check out that version [here](https://rawgit.com/jbt/markdown-editor/mdedit/index.html#Fco9CoAwDAbQ3VMILnZpd8/hJhKK/dCg/SFJ8fri+OBtWjPGh8u9z5dZ0yUEFP/yzQ2Jo69yhl9hFYDmFC2SmvTDusBNKyRzqU89GUpdkYgL/Vfd8AE=)
